tests/pydyad_spsc/run.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+# FLUX: -N 2
+# FLUX: --output=pydyad_spsc_test.out
+# FLUX: --error=pydyad_spsc_test.err
+
+DYAD_INSTALL_LIBDIR="/g/g90/lumsden1/ws/insitu_benchmark/dyad/_test_install/lib"
+KVS_NAMESPACE="pydyad_test"
+CONS_MANAGED_PATH="/l/ssd/lumsden1/pydyad_test"
+PROD_MANAGED_PATH="/l/ssd/lumsden1/pydyad_test"
+NUM_FILES=50
+NUM_INTS=500
+
+export LD_LIBRARY_PATH="$DYAD_INSTALL_LIBDIR:$LD_LIBRARY_PATH"
+
+if [ ! -d "$CONS_MANAGED_PATH" ]; then
+    mkdir -p $CONS_MANAGED_PATH
+fi
+if [ ! -d "$PROD_MANAGED_PATH" ]; then
+    mkdir -p $PROD_MANAGED_PATH
+fi
+
+cat > ./env_file.cons << EOM
+DYAD_PATH_CONSUMER=$CONS_MANAGED_PATH
+DYAD_KVS_NAMESPACE=$KVS_NAMESPACE
+DYAD_DTL_MODE=UCX
+EOM
+
+cat > ./env_file.prod << EOM
+DYAD_PATH_PRODUCER=$PROD_MANAGED_PATH
+DYAD_KVS_NAMESPACE=$KVS_NAMESPACE
+DYAD_DTL_MODE=UCX
+EOM
+
+flux kvs namespace create $KVS_NAMESPACE
+flux exec -r all flux module load $DYAD_INSTALL_LIBDIR/dyad.so \
```

Comment:
This is misleading although I know it works and why you do it in this way.
There could be multiple options we can look into, and I mentioned this to Hari as well.

Option 1: One is to allocate two nodes and explicitly launch modules for producer and consumer to exclusive ranks with corresponding managed paths. You may run commands via `flux exec -r`. However,  the tricky part is that you need to run one of producer and consumer in background if you were to run them concurrently.
Option 2: Create a script for the producer and another one for the consumer. Inside of each script you check if dyad module exists and launch one if not, with the corresponding managed path.

In general, we may assume the same path for consumer and producer. This is the typical use case we envision. However, it does not work with the testing using a single node. That is why you are using producer's managed path.